src/Core/Grid/Data/Factory/StoreGridDataFactory.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\Data\Factory;
+
+use PrestaShop\PrestaShop\Core\Grid\Data\GridData;
+use PrestaShop\PrestaShop\Core\Grid\Data\GridDataInterface;
+use PrestaShop\PrestaShop\Core\Grid\Record\RecordCollection;
+use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteriaInterface;
+
+class StoreGridDataFactory implements GridDataFactoryInterface
+{
+    /**
+     * @var GridDataFactoryInterface
+     */
+    private $storeDataFactory;
+
+    public function __construct(
+        GridDataFactoryInterface $storeDataFactory
+    ) {
+        $this->storeDataFactory = $storeDataFactory;
+    }
+
+    public function getData(SearchCriteriaInterface $searchCriteria): GridDataInterface
+    {
+        $storeData = $this->storeDataFactory->getData($searchCriteria);
+
+        $modifiedRecords = $this->applyModification(
+            $storeData->getRecords()->all()
+        );
+
+        return new GridData(
+            new RecordCollection($modifiedRecords),
+            $storeData->getRecordsTotal(),
+            $storeData->getQuery()
+        );
+    }
+
+    /**
+     * @param array<int, array<string, mixed>> $stores
+     *
+     * @return array<int, array<string, mixed>>
+     */
+    protected function applyModification(array $stores): array
+    {
+        $optionalFields = ['state', 'fax', 'phone'];
+
+        foreach ($stores as $i => $store) {
+            foreach ($store as $key => $value) {
+                if (!in_array($key, $optionalFields, true)) {
+                    continue;
+                }
+
+                // adds placeholder "--" to every field which contains empty value
+                if (!$value) {
```

Comment:
```suggestion
                if (empty($value)) {
```
Should we use empty for a larger check?

Reply:
I don't think so. Now it works perfectly fine and I don't see any case where empty would give any advantage.